doc/feature/restAPI.md

```diff
@@ -0,0 +1,129 @@
+- _registry_ and _schema_ are both keywords that correspond to GET requests.
+  They both need to be handled by the _processGet method of the
+  zeq::http::Server implementation.
+- Add a getRegistry method to the zeq::http::Server implementation that returns
+  the contents of _registrations and _subscriptions containers in the form of
+  a JSON list a tuples defined by a URI and a list of REST operations (currently
+  GET and PUT).
+- This mechanism should be transparent to the user of the class, and should be
+  handled internally to the zeq::http::Server. The schema can be requested at
+  any time on invocation of the getSchema method on the registered object.
+
+## Issues
+### API
+Three options are being discussed regarding the implementation of the API:
+
+#### Option 1: per-object invocation
+GET / returns the list of top-level namespaces, e.g., 'zerobuf', GET
+/zerobuf/render/ returns the objects in zerobuf::render. GET
+/namespace/object/schema returns the schema for the object.
+
+- Example:
+  * Request: ```GET http://hostname:port/```
+  * Response: ```[ "namespace1", "namespace2", "..." ]```
+  * Request: ```GET http://hostname:port/namespace1/```
+  * Response: ```[ "object1", "object2", "..." ]```
+  * Request: ```GET http://hostname:port/namespace1/object1/schema```
+  * Response: ```{ "object1" : { "attribute1": "string", "attribute2": "boolean" } }```
+- Pros: Clean API, and a one-to-one match between a URI and the registered
+  object namespace and name.
+- Cons: Client needs to traverse the tree of URIs to get the list of registered
+  objects. A mechanism to distinguish between an object and a namespace is
+  needed.
+
+#### Option 2: Unique call for all objects
+GET /api-docs returns the list of URIs and schemas for registered objects.
+- Example:
+  * Request: ```GET http://hostname:port/api-docs```
+  * Response:
+```
+{
+  { "namespace/foo": { foo: { "name": string } },
+  { "namespace/bar": { bar: { "x": float, "y": float, "z": float } }
+}
+```
+- Pros: Simple client and server side implementations. No need for a recursive
+  walk in the tree of URIs. Only one invocation is required to get the list of
+  all registered objects.
+- Cons: /api-docs becomes a reserved URI.
+
+#### Option 3: Mix of global registry and per-objet requests
+This option is a mix of option 1 and 2. GET /registry returns the URIs for all
+registered objects, together with the supported REST operation (GET or PUT).
+Each objet can be queried for its schema using the a GET operation on the URI of
+that object, suffixed by /schema.
+
+- Example:
+  - Get URIs to registered objects:
+    * Request: ```GET http://hostname:port/registry```
+    * Response:
+```
+{
+  { "namespace/foo": ["GET", "PUT"] },
+  { "namespace/bar": ["PUT"] }
+}
+```
+  - Get JSON schema for a specific object:
+    * Request: ```GET http://hostname:port/namespace/foo/schema```
+    * Response: ```{ "schema": { "name": string, "age": integer } }```
```

Comment:
In the example above, the JSON response contained "foo" as the top level key. I think that neither "foo" nor "schema" are needed and this should simply return { "name": string, "age": integer }, but if you really want to have a top level key, at least be consistent.